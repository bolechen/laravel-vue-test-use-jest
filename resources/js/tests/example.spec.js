import { createLocalVue, mount } from '@vue/test-utils'
import Vuex from 'vuex'
import ExampleComponent from '@/components/ExampleComponent.vue'

const localVue = createLocalVue()
localVue.use(Vuex)

test('it works', () => {
    expect(1 + 1).toBe(2)
})

describe('ExampleComponent', () => {
    test('test case one', () => {
        const wrapper = mount(ExampleComponent, {
            localVue,
            propsData: {},
            mocks: {},
            stubs: {},
            methods: {}
        })

        expect(wrapper.isVueInstance()).toBeTruthy()
        expect(wrapper).toMatchSnapshot()
    })
})
