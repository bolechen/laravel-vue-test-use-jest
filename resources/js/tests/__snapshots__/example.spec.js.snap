// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ExampleComponent test case one 1`] = `
<div class="container">
  <div class="row justify-content-center">
    <div class="col-md-8">
      <div class="card">
        <div class="card-header">Example Component one</div>
        <div class="card-body">
          I'm an example component.
        </div>
      </div>
    </div>
  </div>
</div>
`;
